api/pkg/resources/data.go

```
@@ -209,6 +211,11 @@ func (d *TemplateData) NodePortRange() string {
 	return d.nodePortRange
 }
 
+// GetMasterVpnAddress panics
+func (d *TemplateData) GetMasterVpnAddress() (string, string, error) {
+	panic("GetMasterVpnAddress not implemented for TemplateData")
```

The panics will all be gone once this is part of a release?

---

That was actually not the plan. Using a `TemplateData` in a usercluster is incorrect and also `UserClusterData` in the clustercontroller is incorrect. Those are mistakes in the code which can only be caused by seriously mixing up userclustercontroller and clustercontroller code paths. In that case I prefer to fail hard and early instead of having a no-op func to satisfy the common interface.